modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/AwsS3ServiceImplTests.java

```diff
@@ -26,11 +31,39 @@
 
 public class AwsS3ServiceImplTests extends ESTestCase {
 
+    private final S3Service.CustomWebIdentityTokenCredentialsProvider webIdentityTokenCredentialsProvider = Mockito.mock(
```

Comment:
should we instantiate this at the beginning of every test? Just in case one test fails and gives unexpected result in subsequent tests?

Reply:
It's not static, so it doesn't get instantiated for every test.

```
2022-01-18T02:26:21,604][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testEndpointSetting] before test
Mock for CustomWebIdentityTokenCredentialsProvider, hashCode: 475523511
[2022-01-18T02:26:21,739][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testEndpointSetting] after test
[2022-01-18T02:26:21,752][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testRepositoryThrottleRetries] before test
Mock for CustomWebIdentityTokenCredentialsProvider, hashCode: 736806917
[2022-01-18T02:26:21,755][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testRepositoryThrottleRetries] after test
[2022-01-18T02:26:21,758][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testAWSDefaultConfiguration] before test
Mock for CustomWebIdentityTokenCredentialsProvider, hashCode: 773823376
[2022-01-18T02:26:21,761][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testAWSDefaultConfiguration] after test
[2022-01-18T02:26:21,764][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testAWSCredentialsDefaultToInstanceProviders] before test
Mock for CustomWebIdentityTokenCredentialsProvider, hashCode: 1848596979
[2022-01-18T02:26:21,772][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testAWSCredentialsDefaultToInstanceProviders] after test
[2022-01-18T02:26:21,775][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testCredentialsIncomplete] before test
Mock for CustomWebIdentityTokenCredentialsProvider, hashCode: 888758980
[2022-01-18T02:26:21,781][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testCredentialsIncomplete] after test
[2022-01-18T02:26:21,785][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testAWSCredentialsFromKeystore] before test
Mock for CustomWebIdentityTokenCredentialsProvider, hashCode: 498506131
[2022-01-18T02:26:21,793][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testAWSCredentialsFromKeystore] after test
[2022-01-18T02:26:21,796][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testSetDefaultCredential] before test
Mock for CustomWebIdentityTokenCredentialsProvider, hashCode: 2102639034
[2022-01-18T02:26:21,799][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testSetDefaultCredential] after test
[2022-01-18T02:26:21,802][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testRepositoryMaxRetries] before test
Mock for CustomWebIdentityTokenCredentialsProvider, hashCode: 1287801154
[2022-01-18T02:26:21,804][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testRepositoryMaxRetries] after test
[2022-01-18T02:26:21,807][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testSupportsWebIdentityTokenCredentials] before test
Mock for CustomWebIdentityTokenCredentialsProvider, hashCode: 1408458911
[2022-01-18T02:26:21,816][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testSupportsWebIdentityTokenCredentials] after test
[2022-01-18T02:26:21,822][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testAWSConfigurationWithAwsSettings] before test
Mock for CustomWebIdentityTokenCredentialsProvider, hashCode: 1298073823
[2022-01-18T02:26:21,824][INFO ][o.e.r.s.AwsS3ServiceImplTests] [testAWSConfigurationWithAwsSettings] after test
```

Reply:
Right, somehow I've read a `static` that does not exist. Sorry for the noise